tuda_workspace_scripts/discovery.py

```diff
@@ -0,0 +1,126 @@
+from .workspace import get_workspace_root
+from robots import DiscoveryServer, load_robots
+from tuda_workspace_scripts import print_warn
+from ros2cli.node.daemon import is_daemon_running, spawn_daemon, shutdown_daemon
+import os
+import xml.etree.ElementTree as ET
+
+
+# dedicated place for temporary files
+tmp_directory = "/tmp/tuda_wss"
+discovery_xml_path = os.path.join(tmp_directory, "discovery_client.xml")
+super_client_xml_path = os.path.join(tmp_directory, "super_client.xml")
+
+def create_discovery_xml(discovery_server_names: list[str]):
+  if not os.path.exists(tmp_directory):
+    os.mkdir(tmp_directory)
+
+  discovery_servers = []
+  robots = load_robots()
+  for name in discovery_server_names:
+    if name == "local_server":
+      discovery_servers.append(DiscoveryServer("127.0.0.1", 11811, "44.53.00.5f.45.50.52.4f.53.49.4d.41"))
+    elif name == "all":
+      for __, robot in robots.items():
+        discovery_servers.extend(robot.discovery_servers)
+      # This is currently unclean because there should be a better toggle/standard
+      discovery_servers.append(DiscoveryServer("127.0.0.1", 11811, "44.53.00.5f.45.50.52.4f.53.49.4d.41"))
+      # No servers can be added anymore, so all other arguments are ignored
+      break
+    else:
+      # Only adding if there is exactly one robot, else the robots xml is probably misformed ore the robot is missing.
+      # check for robots with correct name
+      filtered_robots = []
+      for robot_name, robot_data in robots.items():
+        if robot_name == name:
+          filtered_robots.append(robot_data)
+      if len(filtered_robots) == 1:
+        discovery_servers.extend(filtered_robots[0].discovery_servers)
+      elif len(filtered_robots) == 0:
+        print_warn(f"Couldn't find correct entry for {name} in robot configs. Please check if your selected robot is available.")
+      else:
+        print_warn(f"Found multiple robot entries for {name} in robot configs. Your configs seem to be misformed.")
```

Comment:
This is impossible. robots is a dict. It can't have multiple entries with the same key.